packages/web-forms/src/components/common/map/useMapBlock.ts

```diff
@@ -236,55 +310,69 @@ export function useMapBlock(mode: Mode, events: { onFeaturePlacement: () => void
 
 	const saveCurrentLocation = () => {
 		const location = mapViewControls?.getUserCurrentLocation();
-		if (currentMode.capabilities.canSaveCurrentLocation && location) {
-			loadAndSaveSingleFeature(location.longitude, location.latitude, {
-				[ODK_VALUE_PROPERTY]: formatODKValue(location.longitude, location.latitude),
-			});
+		if (!currentMode.capabilities.canSaveCurrentLocation || !location) {
 			return;
 		}
+
+		const coords = [location.longitude, location.latitude];
+		if (location.altitude != null) {
+			coords.push(location.altitude);
+		}
+
+		if (location.accuracy != null) {
+			coords.push(location.accuracy);
+		}
+
+		const feature = new Feature({
+			geometry: new Point(fromLonLat(coords), 'XYZM'),
```

Comment:
XYZM is only used twice (outside of tests) but it would save me from Googling what it is if it were a constant with a useful name.